addons/hr_calendar/models/calendar_event.py

```diff
@@ -13,14 +13,17 @@ class CalendarEvent(models.Model):
 
     @api.depends('partner_ids', 'start', 'stop', 'allday')
     def _compute_unavailable_partner_ids(self):
-        event_intervals = self._get_events_interval()
-
+        complete_events = self.filtered(
+            lambda event: event.start and event.stop and event.stop >= event.start and event.partner_ids)
+        incomplete_event = self - complete_events
+        for event in incomplete_event:
+            event.unavailable_partner_ids = []
+        if not complete_events:
+            return
+        event_intervals = complete_events._get_events_interval()
```

Comment:
IMO _get_events_interval should implement a watchdog because if a method can take self it should not crash even if you have an emply recordset because if we use it for some reasons else where you'll need to do the same thing over and over again (except of course if the method is designed to crash or explicitly ask for something)

And fir a mater of fact if you return just an empty dict in that case your for loop will just be pass